UIforETW/Settings.cpp

```diff
@@ -143,11 +143,15 @@ void CSettings::OnBnClickedCopystartupprofile()
 	std::wstring source = exeDir_ + fileName;
 
 	wchar_t documents[MAX_PATH];
-	if (!SHGetSpecialFolderPath(*this, documents, CSIDL_MYDOCUMENTS, TRUE))
+
+	const BOOL getMyDocsResult = SHGetSpecialFolderPathW(m_hWnd, documents, CSIDL_MYDOCUMENTS, TRUE);

```

Comment:
I'm not convinced that accessing a protected member variable of the base class directly is better than a conversion operator. But mostly I'm annoyed that m_hWnd is a protected member variable. So much for encapsulation.

At least using the conversion operator doesn't advertise the fact that m_hWnd is completely accessible, and the HWND conversion operator is sure safer than, say, the CString conversion operator to const char*.